### ISSUE TITLE
Transaction error while minting, "Transaction has been reverted by the EVM".

### ISSUE THREAD

**Issue by NishantPacharne** [Shardeum/shardeum-bug-reporting#18]
##### Overview

The contract had already minted 3201 NFT's but now suddenly no mint transaction works, the error I get from web3 is "Transaction has been reverted by the EVM". In remix I tried to mint and got the error "Transaction mined but execution failed"

##### Context 

        -The error is only seen while executing the mint function
        -In fact when a mint transaction takes place, the funds are deducted from wallet but are not received by the contract.

##### Resources

Complete Error, 
`Transaction has been reverted by the EVM: { "blockHash": "0xcec1bc95ed022cd5f0a1f30e6134684d33895a29407a35f47d8abc70148e392d", "blockNumber": 271301, "contractAddress": null, "cumulativeGasUsed": 27737, "data": "0x40c10f19000000000000000000000000f403e8f607420ae290b17b6345bef193d12bf4670000000000000000000000000000000000000000000000000000000000000001", "from": "0xf403e8f607420ae290b17b6345bef193d12bf467", "gasUsed": 27737, "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000", "nonce": "14", "reason": "revert", "status": false, "to": "0xe83a69c8cd50d681895602acdec81f7847e70fde", "transactionHash": "0xcc5bcaa35b81d8ced619345231ffc52e65ba1bc8f78aacc138efcc10444b2ea7", "transactionIndex": 1, "value": "2a802f8630a240000", "events": {} }`

**Comment by MarcusWentz** [comment]
Please add contracts to test with GitHub links and directly in the issue.

Example:

https://github.com/Shardeum/shardeum-bug-reporting/issues/12

Please share the Shardeum Liberty 2.1 explorer for the address and failed transaction.

**Comment by NishantPacharne** [comment]
Sure, here's the contract code,

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.4;

import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
import "@openzeppelin/contracts/security/Pausable.sol";
import "@openzeppelin/contracts/access/Ownable.sol";
import "@openzeppelin/contracts/utils/Counters.sol";

contract LionForceNFT is  ERC721Enumerable, Pausable, Ownable {
    using Counters for Counters.Counter;
    using Strings for uint256;

    Counters.Counter private _tokenIdCounter;

    uint256 public MINT_PRICE = 49 ether;
    string public baseExtension = ".json";
    uint256 public maxSupply = 5000;


    constructor() ERC721("Lion Force NFT", "LFN") {
        _tokenIdCounter.increment();
    }

    function _baseURI() internal pure override returns (string memory) {
        return "ipfs://bafybeib6k2drj6me247gemy5wouaxs2hqusn52dnnkgz22smaqrwyg33h4/";
    }

    function pause() public onlyOwner {
        _pause();
    }

    function unpause() public onlyOwner {
        _unpause();
    }


    function mint(address to, uint256 amount) public payable {
        require(!paused(), "Contract is paused, please try again later");
        require(amount > 0);
        uint256 supply = totalSupply();
        require(supply + amount <= maxSupply);
        require(msg.value >= MINT_PRICE * amount);


        for (uint256 i = 1; i <= amount; i++) {
            _safeMint(to, supply + i);
        }
    }

    function withdraw() public onlyOwner() {
        require(address(this).balance > 0, "Balance is zero");
        payable(owner()).transfer(address(this).balance);
    }



    function tokenURI(uint256 tokenId)
        public
        view
        virtual
        override
        returns (string memory)
    {
        require(
            _exists(tokenId),
            "ERC721Metadata: URI query for nonexistent token"
        );

        string memory currentBaseURI = _baseURI();
        return
            bytes(currentBaseURI).length > 0
                ? string(
                    abi.encodePacked(
                        currentBaseURI,
                        tokenId.toString(),
                        baseExtension
                    )
                )
                : "";
    }
}
```

**Comment by MarcusWentz** [comment]
I tested the contract in Remix IDE in a local network and minting is working.

I tested the same thing in Remix IDE with Liberty 2.1 at address 0xe83a69C8CD50d681895602ACdEC81F7847E70fde. 

The transaction failed:

https://explorer-liberty20.shardeum.org/tx/0x81dd47c8f70e84c2e90819d99272701390155cb17f87f070ab81a2f75e53eb1a

It looks like some of tokenId mint values look like they are not in order:

https://explorer-liberty20.shardeum.org/token/0xe83a69c8cd50d681895602acdec81f7847e70fde

![tokenId](https://user-images.githubusercontent.com/52706599/212109058-c68922c8-19df-42d9-bc0d-fe906f198c64.png)

When I read values in Remix IDE I see:

totalSupply = 3201

ownerOf(3202) =  0x8E8D6eF7F723Fe552D1084e009Da0e56228d1b7A

Have you seen an issue before where the tokenId can get bigger than totalSupply?
